src/Composer/DefinitionMutator.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Nusje2000\ComposerMonolith\Composer;
+
+use LogicException;
+use Symfony\Component\Finder\SplFileInfo;
+
+final class DefinitionMutator implements DefinitionMutatorInterface
+{
+    /**
+     * @var array<mixed, mixed>
+     */
+    protected $originalDefinition;
+
+    /**
+     * @var array<mixed, mixed>
+     */
+    protected $mutatedDefinition;
+
+    /**
+     * @var SplFileInfo
+     */
+    protected $fileInfo;
+
+    public function __construct(SplFileInfo $fileInfo)
+    {
+        $decoded = json_decode($fileInfo->getContents(), true);
+
+        $this->originalDefinition = $decoded;
+        $this->mutatedDefinition = $decoded;
+        $this->fileInfo = $fileInfo;
+    }
+
+    public function setDependency(string $dependency, string $versionConstraint): void
+    {
+        $this->mutatedDefinition['require'][$dependency] = $versionConstraint;
+    }
+
+    public function removeDependency(string $dependency): void
+    {
+        unset($this->mutatedDefinition['require'][$dependency]);
+    }
+
+    public function setDevDependency(string $dependency, string $versionConstraint): void
+    {
+        $this->mutatedDefinition['require-dev'][$dependency] = $versionConstraint;
```

Comment:
What if this dependency is already set with a different version constraint?

Maybe throw and exception if already set (if so, don't forget to add `@throws` to the interface).
If you want to overwrite without warning/error, I would document this in the interface.

Reply:
The behavior of overriding dependencies that are already present is expected. This is used to update versions of dependencies and add missing.